test/main_dialog_spec.rb

```diff
@@ -191,18 +191,74 @@ def expect_update_table_with(expected_items)
   end
 
   describe "#cancel_handler" do
-    before do
-      mock_ui_events(:cancel)
-    end
+    context "if there are any change" do
+      let(:alternative_dialog) { double("AlternativeDialog") }
 
-    it "doesn't save any change" do
-      expect_any_instance_of(UpdateAlternatives::Alternative).to_not receive(:save)
-      dialog.run
+      before do
+        mock_ui_events(:edit_alternative, :cancel)
+        allow(Yast::UI).to receive(:QueryWidget)
+          .with(:alternatives_table, :CurrentItem)
+          .and_return(0)
+        allow(UpdateAlternatives::AlternativeDialog).to receive(:new)
+          .and_return(alternative_dialog)
```

Comment:
to be honest I do not get advantage of using empty double for alternative_dialog.
It think it will be better to have
```
allow(UpdateAlternatives::AlternativeDialog).to receive(:new)
  .and_return(double("AlternativeDialog", run: false))
```
so next line is not needed and more readable from my POV